tree/ntuple/v7/test/ntuple_evolution.cxx

```diff
@@ -0,0 +1,251 @@
+#include "ntuple_test.hxx"
+
+#include <TInterpreter.h>
+
+#include <functional>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+#include <unistd.h>
+#include <sys/wait.h>
+
+namespace {
+// These helpers are split into an *Impl function and a macro to check for and propagate fatal failures to the caller.
+
+void WriteOldInForkImpl(std::function<void(void)> old)
+{
+   pid_t pid = fork();
+   if (pid == -1) {
+      FAIL() << "fork() failed";
+   }
+
+   if (pid == 0) {
+      old();
+      if (::testing::Test::HasFatalFailure())
+         exit(EXIT_FAILURE);
+      exit(EXIT_SUCCESS);
+   }
+
+   int wstatus;
+   if (waitpid(pid, &wstatus, 0) == -1) {
+      FAIL() << "waitpid() failed";
+   } else if (!WIFEXITED(wstatus) || WEXITSTATUS(wstatus) != 0) {
+      FAIL() << "child did not exit successfully";
+   }
+}
+
+#define WriteOldInFork(old)                   \
+   do {                                       \
+      WriteOldInForkImpl(old);                \
+      if (::testing::Test::HasFatalFailure()) \
+         return;                              \
+   } while (0)
+
+void DeclarePointerImpl(std::string_view type, std::string_view variable, void *ptr)
+{
+   auto ptrString = std::to_string(reinterpret_cast<std::uintptr_t>(ptr));
+   std::ostringstream ptrDeclare;
+   ptrDeclare << type << " *" << variable << " = ";
+   ptrDeclare << "reinterpret_cast<" << type << " *>(" << ptrString << ");";
+   ASSERT_TRUE(gInterpreter->Declare(ptrDeclare.str().c_str()));
+}
+
+#define DeclarePointer(type, variable, ptr)    \
+   do {                                        \
+      DeclarePointerImpl(type, variable, ptr); \
+      if (::testing::Test::HasFatalFailure())  \
+         return;                               \
+   } while (0)
+
+void EvaluateIntImpl(const char *expression, int *value)
+{
+   auto interpreterValue = gInterpreter->MakeInterpreterValue();
+   ASSERT_TRUE(gInterpreter->Evaluate(expression, *interpreterValue));
+   *value = interpreterValue->GetAsLong();
+}
+
+#define EXPECT_EVALUATE_EQ(expression, expected) \
+   do {                                          \
+      int _value;                                \
+      EvaluateIntImpl(expression, &_value);      \
+      if (::testing::Test::HasFatalFailure())    \
+         return;                                 \
+      EXPECT_EQ(expected, _value);               \
+   } while (0)
+} // namespace
+
+TEST(RNTupleEvolution, RemovedMember)
+{
+   FileRaii fileGuard("test_ntuple_evolution_removed_member.root");
+
+   WriteOldInFork([&] {
+      // The child process writes the file and exits, but the file must be preserved to be read by the parent.
+      fileGuard.PreserveFile();
+
+      ASSERT_TRUE(gInterpreter->Declare(R"(
+struct RemovedMember {
+   int fInt1 = 1;
+   int fInt2 = 2;
+   int fInt3 = 3;
+};
```

Comment:
Do we need (I think not) / want (not sure) `ClassDef` declarations?

Reply:
Both ways may be useful, I think. Without to check that we don't depend on the class version. With `ClassDef` to check warnings / reporting, if applicable.